addons/project/models/project_task.py

```diff
@@ -1085,13 +1085,17 @@ def create(self, vals_list):
         # (portal) users that don't have write access can still create a task
         # in the project that will be checked using record rules
         new_context["default_create_in_project_id"] = default_project_id
+        is_portal = self.env.user._is_portal() and not self.env.su
+        if is_portal:
+            default_milestone_id = new_context.pop('default_milestone_id', False)
```

Comment:
```suggestion
        if not self._has_field_access(self._fields['milestone_id'], 'write'):
            # remove milestone_id if we have no access to it
            new_context.pop('milestone_id', False)
```
You can add after user_ids